app/assets/javascripts/admin/bulk_order_management.js.coffee

```diff
@@ -162,6 +163,14 @@ angular.module("ofn.admin").controller "AdminOrderMgmtCtrl", [
       $scope.supplierFilter = $scope.suppliers[0].id
       $scope.orderCycleFilter = $scope.orderCycles[0].id
       $scope.quickSearch = ""
+
+    $scope.weightAdjustedPrice = (lineItem, oldValue) ->
+      if oldValue <= 0
+        oldValue = lineItem.units_variant.unit_value
+      if lineItem.unit_value <= 0
+        lineItem.unit_value = lineItem.units_variant.unit_value
+      lineItem.price = lineItem.price * lineItem.unit_value / oldValue
+      #$scope.bulk_order_form.line_item.price.$setViewValue($scope.bulk_order_form.line_item.price.$viewValue)
 ]
 
```

Comment:
Hmmmmm, it makes me uncomfortable. What I am advocating a migration for existing AND some logic that sets the unit_value of every LineItem as it is created, so that we maintain data integrity. Sorry to be a pain, my issue is that the logic of "if the unit_value of the line_item is zero then use the unit_value of the variant, otherwise unit_value of the line_item" seems overly complicated and means that we will require knowledge of this particular pattern in every instance that we want to use the unit_value of a line_item. I can't really see the logic in storing a unit_value of zero, when that is not the unit value. If we are going to go to the trouble of having a field for unit_value on line_item, I think it may as well just be populated with the actual unit_value, and then our interactions with it can be intuitive, consistent and error free.